src/main/java/htsjdk/samtools/util/AbstractLocusIterator.java

```diff
@@ -176,8 +171,14 @@ public AbstractLocusIterator(final SamReader samReader, final IntervalList inter
         this.samReader = samReader;
         this.useIndex = useIndex;
         if (intervalList != null) {
-            intervals = intervalList.uniqued().getIntervals();
-            this.referenceSequenceMask = new IntervalListReferenceSequenceMask(intervalList);
+            if (!intervalList.getHeader().getSequenceDictionary().isSameDictionary(getHeader().getSequenceDictionary())) {
+                throw new SAMException("The sequence dictionary of the interval list file differs from the sequence dictionary of the input SAM file.");
+            }
+
+            final boolean isSorted = intervalList.getHeader().getSortOrder() == SAMFileHeader.SortOrder.coordinate;
```

Comment:
Yes good question. First of all, yes, we need both the SAM file and the interval list file to be coordinate sorted. AbstractLocusIterator logs a warning if the sort order is unspecified, and throws an exception if it is specified but not coordinate. What I'm doing here is checking if the interval list is coordinate sorted, because if it is, we might as well skip the sorting. Or do you think we should sort either way?